### ISSUE TITLE
IDISP003 should not warn when early return

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#95]
```cs
namespace RoslynSandbox
{
    using System.IO;

    public class Foo
    {
        private static bool TryGetStream(string fileName, out Stream stream)
        {
            if (File.Exists(fileName))
            {
                stream = File.OpenRead(fileName);
                return true;
            }

            stream = null;
            return false;
        }
    }
}
```
